### PR TITLE
client/asset/btc: work around PublishTransaction delays

### PR DESCRIPTION
**Pull request by chappjc**
Because of (a) what appears to be a neutrino bug, with a fix proposed in https://github.com/lightninglabs/neutrino/pull/240, and (b) that neutrino `wallet.PublishTransaction` is blocking while `inv` requests go to peers and `getdata` responses come back, this PR attempts to avoid an unnecessary delay with BTC SPV when broadcasting transactions.

In all my observations, `PublishTransaction` uses the **entire** `neutrino.Config.BroadcastTimeout`, which is nearly an intolerable delay for many operations.

With neutrino as the wallet's chain service, its `chainClient.SendRawTransaction` call is blocking for up to `neutrino.Config.BroadcastTimeout` while peers either respond to the `inv` request with a `getdata`, or time out. However, in virtually all cases, we just need to know that btcwallet was able to create and store the transaction record, and pass it to the chain service.  This is a yucky workaround to consider until https://github.com/lightninglabs/neutrino/pull/240 is accepted, maybe even complementary to it.

### CODE REVIEW

**Comment by chappjc** [comment]
The upstream fix is unlikely to happen this week, and even if it did, we would definitely not want broadcast to consistently block for even 4 seconds, which is appears to be most common even when all peers to respond with `getdata`.  Going to wrap this up and fix the tests, but I have been testing regularly in the past week.

In parallel, I'm looking to make certain requests async, with particular focus on not holding up preimage and match_proof requests.

File: client/asset/btc/spv.go [hunk 1]
```diff
@@ -1160,12 +1195,17 @@ func (w *spvWallet) startWallet() error {
...
+		Database:      w.neutrinoDB,
+		ChainParams:   *w.chainParams,
+		PersistToDisk: true, // keep cfilter headers on disk for efficient rescanning
+		AddPeers:      addPeers,
+		ConnectPeers:  w.connectPeers,
+		// WARNING: PublishTransaction currently uses the entire duration
+		// because if an external bug, but even if the resolved, a typical
+		// inv/getdata round trip is ~4 seconds, so we set this so neutrino does
+		// not cancel queries too readily.
+		BroadcastTimeout: 6 * time.Second,
```
Comment (buck54321) [inline client/asset/btc/spv.go:1208]: Looks like the default is 5? So we're actually giving neutrino **more** time.

**Review by buck54321 — APPROVED** [review]

File: client/asset/btc/spv.go (hunk 1 above)
Comment (chappjc) [reply client/asset/btc/spv.go:1208]: Correct.  So 10 or even 30 would be fine with this approach, but I figured I would set it to something based on my empirical observations now that I've studies it more in practice.

**Comment by chappjc** [comment]
We will definitely see a follow up to this in the future in the form of a neutrino update and/or a new btcwallet method that provides partially asynchronous PublishTransaction functionality without waiting for peer responses, only tx recording and queuing.

**Pull request merged**